### PR TITLE
[MM-42584, MM-42604] post Actions Menu merge fixes

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR fixes several discovered issues that were discovered after the Actions Menu was merged. https://github.com/mattermost/mattermost-webapp/pull/8853

### Broken Cypress test introduced by "More Actions Menu Restructure"
- [x] https://mattermost.atlassian.net/browse/MM-42584 

broken cypress tests from this PR merge https://github.com/mattermost/mattermost-webapp/pull/8853.  I fixed the e2e and unit tests that seemed to be related to the Actions Menu change.  

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42604

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/8853.

#### Screenshots
n/a

### CODE REVIEW

**Comment by jfrerich** [comment]
All, this PR is trying to make it to the March 30th release. 

@saturninoabril, this PR solves what I believe are the issues caused by the new Actions Menu.  I tested the failing tests individually from this run. Can you kick of a full e2e test to see that this branch resolves those issues?

**Comment by Phrynobatrachus** [comment]
I see two failures on the e2e run from

`tests/integration/accessibility/accessibility_post_spec.js`

when I run these locally the actions button doesn't seem to show up in the hover menu? No idea why, `PluginsEnabled` is true which as far as I can tell is the only thing that should control whether or not it does show.

**Comment by jfrerich** [comment]
@Phrynobatrachus, the only other condition is the you are not on a system message. I just tested locally again it passes on my end.  If you can repro this, maybe we can zoom so I can debug it.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @jfrerich!
- MM-42576 - new category dialog verified working
- MM-42636 - reply and mark as unread verified working
- MM-42584 - thanks and great work in resolving failed tests! See couple of comments.
- MM-42604 - status aligned with its icon to the right but left alignment is not with the do not disturb (see screenshot). Do you know if it's related to changes with actions menu?
Screen Shot 2022-03-23 at 2 27 08 PM

**Comment by saturninoabril** [comment]
Added @DHaussermann for thorough QA review.

**Review by larkox — APPROVED** [review]
LGTM

File: e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js [hunk 1]
```diff
@@ -201,12 +201,16 @@ describe('Verify Accessibility Support in Post', () => {
                 cy.get(`#CENTER_flagIcon_${postId}`).should('be.focused').and('have.attr', 'aria-label', 'save');
                 cy.focused().tab();
 
+                // * Verify focus is on the actions button
+                cy.get(`#CENTER_actions_button_${postId}`).should('be.focused').and('have.attr', 'aria-label', 'actions');
+                cy.focused().tab();
+
```
[the remark this answers is not in the document]
Comment (jfrerich) [reply e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js]: @saturninoabril, I added a check for plugins enabled which will run or skip tihs check based on the config setting.  see e827894

image

File: e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js [hunk 2]
```diff
@@ -251,6 +255,10 @@ describe('Verify Accessibility Support in Post', () => {
                     cy.focused().tab({shift: true});
                 }
 
+                // * Verify focus is on the actions button
+                cy.get(`#RHS_COMMENT_actions_button_${postId}`).should('be.focused').and('have.attr', 'aria-label', 'actions');
+                cy.focused().tab({shift: true});
+
```
[the remark this answers is not in the document]
Comment (jfrerich) [reply e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js]: same as above solutions.  only check run this is pluginsenabled

File: e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfrerich) [reply e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js]: Let me know if this runs correctly on your end

**Comment by jfrerich** [comment]
I checked an older commit and I believe I did introduce this issue. Working on it now.
image

File: e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/tests/integration/accessibility/accessibility_post_spec.js]: Same failure - https://mattermost-cypress-report.s3.amazonaws.com/174218-f9a851c-onprem-ent-server-pr-19812-mattermost/mm-ee-testf9a851c/mochawesome.html

I think it depends on whether there is apps or plugin loaded with pluggable items.

I'm fine to remove such check or run the test with new custom admin user (`cy.apiCreateCustomAdmin()`)

**Comment by jwilander** [comment]
@jfrerich is it possible to pull out the fix for https://mattermost.atlassian.net/browse/MM-42636 and get that merged sooner? I think that one's affecting some people's day-to-day on community.

**Comment by jfrerich** [comment]
@jwilander, let work on that and see if I can get that in quicker.

**Comment by amyblais** [comment]
@jfrerich Is this ready for re-reviews?

**Comment by jfrerich** [comment]
@amyblais, Yes this is ready for re-review.  I have pulled out the code that was moved to other smaller PRs. THis PR now focuses only on the previous failing e2e tests.

**Comment by jfrerich** [comment]
@saturninoabril, to keep this PR small, I removed all code that was going to fix non e2e related tests. I've also addressed your latest PR feedback. I'd like to get this merged and if any issues are discovered during testing, we'll craete a ticket and solve them separately!  Let me know if you have any other requests for changes!

**Comment by saturninoabril** [comment]
@jfrerich Sounds good to me. Please make the build passing then we'll run E2E.

**Comment by jfrerich** [comment]
@saturninoabril, the build is now passing

**Review by saturninoabril — APPROVED** [review]
Thanks @jfrerich! Tested and passed.
- `MM-T1225` - did local test and passed as well. Not related to this PR but we'll watch out separately.

**Comment by saturninoabril** [comment]
@jfrerich Feel free to merge.

**Pull request merged**